### DISCUSSION TITLE
how to process messages from users

### DISCUSSION THREAD

**Discussion by tokensave** [def-studio/telegraph#646] [Q&A]
I have action buttons in Handler
public function addNumbers()
{
$this->chat
->message('Choose a provider')
->keyboard(
Keyboard::make()->buttons(
[
Button::make('🔵 Telegram')->action('addNumbersSalesmen')->param('name', 1),
Button::make('   WhatsApp')->action('addNumbersSalesmen')->param('name', 2),
])
)
->send();
}
How can I add the ability for the client to write his number for processing in the addNumbersSalesmen function?
I can’t yet figure out where this is in the documentation
please tell me
I write in Laravel framework

**Comment by fabio-ivona** [comment]
Hi, you should use a webhook to accomplish what you are saying 

https://docs.defstudio.it/telegraph/v1/webhooks/overview
